src/tinygp/kernels/quasisep.py

```diff
@@ -737,6 +741,177 @@ def transition_matrix(self, X1: JAXArray, X2: JAXArray) -> JAXArray:
         ).real
 
 
+@dataclass
+class carma(Quasisep):
+    r"""A continuous autoregressive moving average (CARMA) process
+
+    This process has the power spectrum
+
+    .. math::
+
+        P(\omega) = \sigma^2\,\frac{\sum_{q} \beta_q\,(i\,\omega)^q}{\sum_{p}
+            \alpha_p\,(i\,\omega)^p}
+
+    defined following `Kelly et al. (2014) <https://arxiv.org/abs/1402.5978>`_.
+
+    Unlike other kernels, this *must* be instatiated using the :func:`init`
+    method instead of the usual constructor:
+
+    .. code-block:: python
+
+        kernel = CARMA2.init(log_alpha=..., log_beta=...)
+    """
+    alpha: JAXArray
+    beta: JAXArray
+    arroots: JAXArray
+    acf: JAXArray
+    real_mask: JAXArray
+    complex_mask: JAXArray
+    complex_select: JAXArray
+    obsmodel: JAXArray
+
+    @classmethod
+    def init(cls, log_alpha: JAXArray, log_beta: JAXArray) -> "carma":
```

Comment:
To be consistent with the style of `tinygp`, this should take `alpha` and `beta` (not `log_` as input).

Reply:
Understood. I can remove the 'log' part, but for any CARMA process to be stationary, we need alphas/betas to be positive.  In short, negative values of alphas/betas return an 'invalid' model. Maybe I should add an assertion to validate the input?

Reply:
This kind of constraint is common across most of the kernels implemented in `tinygp`, but the design decision was to just clearly document the constraints in the docstrings, rather than trying to reparameterize everything in terms of `log`, in particular because different people/modeling frameworks have different approaches to enforcing these kinds of constraints. It then becomes the user's responsibility to make sure that they're constraining their parameters appropriately!